src/Components/Server/src/Circuits/CircuitRegistry.cs

```diff
@@ -189,12 +189,12 @@ public virtual async Task<CircuitHost> ConnectAsync(string circuitId, IClientPro
                 // Transition the host from disconnected to connected if it's available. In this critical section, we return
                 // an existing host if it's currently considered connected or transition a disconnected host to connected.
                 // Transfering also wires up the client to the new set.
-                (circuitHost, previouslyConnected) = ConnectCore(circuitId, clientProxy, connectionId);
+                (circuitHost, previouslyConnected) = ConnectCore(circuitSecret, clientProxy, connectionId);
 
                 if (circuitHost == null)
                 {
-                    Log.FailedToReconnectToCircuit(_logger, circuitId);
-                    // Failed to connect. Nothing to do here.
+                    Log.FailedToFindCircuit(_logger, circuitSecret);
```

Comment:
I gave this it's own logging that's more specific. Note that it's TRACE because the secret is involved.